### PR TITLE
Only apply opacity to bar if there is a raw level of detail channel

### PR DESCRIPTION
**Pull request by kanitw**
Part of #2027

### CODE REVIEW

**Comment by kanitw** [comment]
vega_editor

File: src/compile/mark/init.ts [hunk 1]
```diff
@@ -50,7 +52,9 @@ function defaultOpacity(mark: Mark, encoding: Encoding, stacked: StackProperties
     }
   }
   if (mark === BAR && !stacked) {
-    if (channelHasField(encoding, 'color') || channelHasField(encoding, 'detail') || channelHasField(encoding, 'size')) {
+    if (some(LEVEL_OF_DETAIL_CHANNELS, (channel) => {
+      return hasRawField(encoding, channel);
```
Comment (domoritz) [inline src/compile/mark/init.ts]: return boolean?

File: src/compile/mark/init.ts [hunk 2]
```diff
@@ -60,6 +64,21 @@ function defaultOpacity(mark: Mark, encoding: Encoding, stacked: StackProperties
   return undefined;
 }
 
+function hasRawField(encoding: Encoding, channel: Channel) {
+  const channelDef = encoding[channel];
+  if (isArray(channelDef)) { // detail
+    for (const fieldDef of channelDef) {
```
Comment (domoritz) [inline src/compile/mark/init.ts]: `return some(channelDef, fielDef => fieldDef.aggregate)`?

Comment (kanitw) [inline src/compile/mark/init.ts]: * !fieldDef.aggregate

Comment (domoritz) [inline src/compile/mark/init.ts]: Yes

**Pull request merged**